src/main/java/codeu/model/store/basic/ConversationStore.java

```diff
@@ -117,6 +118,16 @@ public Conversation getConversationWithTitle(String title) {
     return null;
   }
 
+  /** Find and return the Conversation with the given Id. */
+  public Conversation getConversationWithID(UUID id) {
+    for (Conversation conversation : conversations) {
```

Comment:
Since this is an O(n) algorithm, and this is only used when we want to find the actual `Conversation` objects associated with a specific `Tag`, why don't we just change the `Tag` constructor and `Tag.addConversation()` to take in a `Conversation` object directly instead of the UUID of a `Conversation`?

Reply:
When we put a `Tag` entity into datastore, each attribute has to be a string/certain other values (we wouldn't be able to set an entity attribute to a `Conversation` object). I guess we could create a wrapper method so that we could have `Tag` take in a `Conversation` object? But that might be more work for what it's worth. LMK what you think!

Reply:
Also check out the PR 73 that gets ride of methods like `addConversation` to Tags and such